halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/config/v1/ResourceConfig.java

```diff
@@ -17,13 +17,49 @@
 package com.netflix.spinnaker.halyard.config.config.v1;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
 import org.springframework.stereotype.Component;
 import org.yaml.snakeyaml.DumperOptions;
 import org.yaml.snakeyaml.Yaml;
 
 @Component
 public class ResourceConfig {
+  /**
+   * Path to where the halconfig file is located.
+   *
+   * @param path Defaults to ~/.hal/config.
+   * @return The path with home (~) expanded.
+   */
+  @Bean
+  String halconfigPath(@Value("${halconfig.filesystem.halconfig:~/.hal/config}") String path) {
+    return replaceBashHome(path);
+  }
+
+  /**
+   * Version of halyard.
+   *
+   * This is useful for implementing breaking version changes in Spinnaker that need to be migrated by some tool
+   * (in this case Halyard).
+   *
+   * @param version is the version (seems like the i.d. function for Spring Boot).
+   * @return the version of halyard.
+   */
+  @Bean
+  String halyardVersion(@Value("${Implementation-Version:unknown}") String version) {
```

Comment:
Yes. I was assuming you are reflecting here (Implementation-Version is for halyard runtime, not something you are imposing on it from the outside). If so, then halyard is running from a build where gradle packaged this module with a version. getPacakge().getImplementationVersion() will be that version from Java's perspective. Note it is "Package" not "Packer". I assume you have a typo and did not misread.

At least, that is the empirical evidence I have from trying to look at versions from a kork library. I went that route to get the kork version (which I can argue makes more sense for what I wanted) where the spring property was rarely defined in current microservices.

You can go the above route, just be aware that you'll have to change how they are currently built for this to work. Since you ask for "halyard version" specifically, I think you want the halyard package version and not the microservice runtime version. But that might not be your intent, in which case dont call this the "halyard" version.